x/crosschain/keeper/v2_zevm_inbound.go

```diff
@@ -32,66 +47,87 @@ func (k Keeper) ProcessZEVMInboundV2(
 			value = big.NewInt(0)
 			receiver = callEvent.Receiver
-			contractAddress = callEvent.Raw.Address
+			callOptions = callEvent.CallOptions
+			receiverChainID = big.NewInt(0) // Receiver chain ID is only used for withdraws when coin type is ZETA.
 		} else {
 			zrc20 = withdrawalAndCallEvent.Zrc20
 			value = withdrawalAndCallEvent.Value
 			receiver = withdrawalAndCallEvent.Receiver
-			contractAddress = withdrawalAndCallEvent.Raw.Address
+			receiverChainID = withdrawalAndCallEvent.ChainId
+			callOptions = withdrawalAndCallEvent.CallOptions
 		}
 
-		// get several information necessary for processing the inbound
-		foreignCoin, found := k.fungibleKeeper.GetForeignCoins(ctx, zrc20.Hex())
-		if !found {
-			ctx.Logger().
-				Info(fmt.Sprintf("cannot find foreign coin with contract address %s", contractAddress.Hex()))
-			return nil
-		}
-		receiverChain, found := k.zetaObserverKeeper.GetSupportedChainFromChainID(ctx, foreignCoin.ForeignChainId)
-		if !found {
+		wzetaContractAddress, err := k.fungibleKeeper.GetWZetaContractAddress(ctx)
+		if err != nil {
 			return errorsmod.Wrapf(
-				observertypes.ErrSupportedChains,
-				"chain with chainID %d not supported",
-				foreignCoin.ForeignChainId,
+				types.ErrCannotProcessWithdrawal,
+				"failed to get WZeta contract address: %v", err,
 			)
 		}
-		gasLimitQueried, err := k.fungibleKeeper.QueryGasLimit(
-			ctx,
-			ethcommon.HexToAddress(foreignCoin.Zrc20ContractAddress),
-		)
+
+		var inboundDetails InboundDetails
+		// The following condition checks if the withdrawal is for ZETA or ZRC20.
+		// ZRC20 cointype can be further classified as ZRC20 or Gas , based on foreign coin or NoAssetCall.
+		// Note: NoAssetCall is not supported for ZETA
+		switch {
+		case zrc20 == wzetaContractAddress:
+			inboundDetails, err = k.getZetaInboundDetails(ctx, receiverChainID, callOptions)
+		default:
+			inboundDetails, err = k.getErc20InboundDetails(ctx, zrc20, callEvent != nil)
+		}
+
 		if err != nil {
-			return err
+			return errorsmod.Wrapf(
+				types.ErrInvalidWithdrawalEvent,
+				"failed to parse inbound details for withdraw: %v", err,
+			)
 		}
 
-		// validate data of the withdrawal event
-		coinType := foreignCoin.CoinType
-		if callEvent != nil {
-			coinType = coin.CoinType_NoAssetCall
+		// if the coin type is ZETA, we need to burn the coins as the GatewayZEVM contract transfers the ZETA to the fungible module account.
+		// For ERC20 and GAS coin types; the GatewayZEVM directly burns the tokens, so we can skip this step.
+		// NOTE: value does not include tokens paid for the gas which is burned by the GatewayZEVM contract for both cases
+		if inboundDetails.coinType == coin.CoinType_Zeta {
+			err := k.bankKeeper.BurnCoins(ctx,
+				fungibletypes.ModuleName,
+				sdk.NewCoins(sdk.NewCoin(config.BaseDenom, sdkmath.NewIntFromBigInt(value))))
+			if err != nil {
+				return errorsmod.Wrapf(
+					types.ErrInvalidWithdrawalEvent,
+					"failed to burn ZETA coins: %v", err,
+				)
+			}
```

Comment:
added burnig of ZETA tokens as they have already been trasnafered to the fungible module